packages/instant-apps-components/src/components/instant-apps-export/instant-apps-export.tsx

```diff
@@ -611,4 +631,152 @@ export class InstantAppsExport {
       }
     }
   }
+
+  createScreenshot(): void {
+    if (this.view != null) {
+      this.screenshotPreview = document.createElement('div');
+      this.screenshotPreview.className = 'screenshot-preview hide';
+      this.screenshotImgContainer = document.createElement('div');
+      this.screenshotImgContainer.className = 'screenshot-img-container';
+      this.screenshotImg = document.createElement('img');
+      const screenshotBtnContainer = document.createElement('div');
+      const exportBtn = document.createElement('calcite-button');
+      const returnBtn = document.createElement('calcite-button');
+      exportBtn.innerHTML = this.messages?.export;
+      returnBtn.innerHTML = this.messages?.returnToMap;
+      returnBtn.appearance = 'outline-fill';
+      exportBtn.onclick = this.exportOnClick.bind(this);
+      returnBtn.onclick = this.screenshotReturn.bind(this);
+      screenshotBtnContainer.append(returnBtn, exportBtn);
+      this.screenshotImgContainer.append(this.screenshotImg, screenshotBtnContainer);
+      this.screenshotPreview.append(this.screenshotImgContainer);
+      this.view.container.append(this.screenshotPreview);
+    }
+  }
+
+  createMaskDiv(): void {
+    if (this.view != null) {
+      this.maskDivEl = document.createElement('div');
+      this.maskDivEl.id = 'screenshot-mask';
+      this.maskDivEl.className = 'hide screenshot-cursor';
+      const maskStylingEl = document.createElement('style');
+      maskStylingEl.id = 'maskStyling';
+      maskStylingEl.innerHTML = maskStyling;
+      this.view.container.append(maskStylingEl);
+      this.view.container.append(this.maskDivEl);
+    }
+  }
+
+  screenshotReturn(): void {
+    this.screenshotPreview.classList.add('hide');
+    this.exportIsLoading = false;
+    this.screenshot = null;
+  }
+
+  handleScreenshot(): void {
+    if (this.view == null) return;
+    this.exportIsLoading = true;
+    this.createMaskDiv();
+    this.createScreenshot();
+    this.view.container.classList.add('screenshot-cursor', 'relative');
+    let area: __esri.MapViewTakeScreenshotOptionsArea | __esri.MapViewTakeScreenshotOptionsArea | undefined;
+
+    const dragHandler = this.view.on('drag', async event => {
+      if (this.view == null) return;
+      event.stopPropagation();
+      if (event.action !== 'end') {
```

Comment:
Is this the logic where the drawing of the mask div is started/finished?

can the logic in the if/else blocks be broken up into separate methods, with method names that describe what each block is doing?